ai_review_test/prompt_builder.py

```
@@ -0,0 +1,50 @@
+                        f'"{item.get("label")}": {item.get("description", "")}'
+                        for item in items if "label" in item
+                    ]
+                    hint_block = f'- {key} options:\n  ' + "\n  ".join(options)
+                    hints.append(hint_block)
+        return "\n".join(hints)
+
+    def build(self, user_input: str, schema_json: str, schema_dict: dict) -> str:
+        try:
+            schema_obj = json.loads(schema_json)
+        except json.JSONDecodeError:
+            raise ValueError("Invalid schema JSON provided")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)